### PR TITLE
OATK module implementation

### PR DESCRIPTION
**Pull request by ksenia-krasheninnikova**
Implements 
https://github.com/nf-core/modules/issues/4591

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by ksenia-krasheninnikova** [comment]
PR for plastid test dataset: https://github.com/nf-core/test-datasets/pull/1059

File: modules/nf-core/oatk/main.nf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/oatk:1.0':
+        'biocontainers/oatk:1.0' }"
+
+    input:
+    tuple val(meta), path(reads)
+    tuple path(mito_hmm), path(mito_hmm_h3f), path(mito_hmm_h3i), path(mito_hmm_h3m), path(mito_hmm_h3p)
+    tuple path(pltd_hmm), path(pltd_hmm_h3f), path(pltd_hmm_h3i), path(pltd_hmm_h3m), path(pltd_hmm_h3p)
+    path(tmp)
```
Comment (mahesh-panchal) [inline modules/nf-core/oatk/main.nf:14]: Usually tmp folders don't need to be staged. Is there a necessity for staging the tmp folder?

**Review by mahesh-panchal — COMMENTED** [review]
Could you convert this module to the latest format please with nf-test?

File: modules/nf-core/oatk/main.nf (hunk 1 above)
Comment (ksenia-krasheninnikova) [reply modules/nf-core/oatk/main.nf:14]: Good point. tmp can be specified in args if required.

**Comment by mahesh-panchal** [comment]
Also as a reminder `nf-core` tools has the option to migrate tests from pytest to nf-test.
```
nf-core modules create <tool>/<subtool> --migrate-pytest
```

File: modules/nf-core/oatk/main.nf [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        $args \\
+        $mito_hmm_arg \\
+        $pltd_hmm_arg \\
+        -t $task.cpus \\
+        -o ${prefix} \\
+        $reads
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        oatk : \$(echo \$(oatk --version 2>&1) )
```
Comment (mahesh-panchal) [inline modules/nf-core/oatk/main.nf:56]:
```suggestion
        oatk : \$(oatk --version 2>&1)
```

File: tests/modules/nf-core/oatk/main.nf [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
```
Comment (mahesh-panchal) [inline tests/modules/nf-core/oatk/main.nf]: These files under `tests/` should go.

**Review by mahesh-panchal — APPROVED** [review]

**Pull request merged**